hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DataStreamer.java

```diff
@@ -937,13 +935,13 @@ void waitForAckedSeqno(long seqno) throws IOException {
             try {
               dataQueue.wait(1000); // when we receive an ack, we notify on
               long duration = Time.monotonicNowNanos() - begin;
-              if (duration > writeTimeout * NANOSECONDS_PER_MILLISECOND) {
+              if (TimeUnit.NANOSECONDS.toMillis(duration) > writeTimeout) {
                 LOG.error("No ack received, took {}ms (threshold={}ms). "
                     + "File being written: {}, block: {}, "
                     + "Write pipeline datanodes: {}.",
-                    duration / NANOSECONDS_PER_MILLISECOND, writeTimeout, src, block, nodes);
+                    TimeUnit.NANOSECONDS.toSeconds(duration), writeTimeout, src, block, nodes);
```

Comment:
TimeUnit.NANOSECONDS.toMillis(duration) ？

Reply:
Fixed it, thanks sir.